cart/infrastructure/InMemoryBehaviour.go

```diff
@@ -99,23 +99,66 @@ func (cob *InMemoryBehaviour) DeleteItem(ctx context.Context, cart *domaincart.C
 }
 
 // UpdateItem updates a cart item
-func (cob *InMemoryBehaviour) UpdateItem(ctx context.Context, cart *domaincart.Cart, itemID string, deliveryCode string, itemUpdateCommand domaincart.ItemUpdateCommand) (*domaincart.Cart, domaincart.DeferEvents, error) {
+func (cob *InMemoryBehaviour) UpdateItem(ctx context.Context, cart *domaincart.Cart, itemUpdateCommand domaincart.ItemUpdateCommand) (*domaincart.Cart, domaincart.DeferEvents, error) {
 	if !cob.cartStorage.HasCart(cart.ID) {
 		return nil, nil, fmt.Errorf("cart.infrastructure.InMemoryBehaviour: Cannot add - Guestcart with id %v not existent", cart.ID)
 	}
 
+	err := cob.updateItem(ctx, cart, itemUpdateCommand)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = cob.cartStorage.StoreCart(cart)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "cart.infrastructure.InMemoryBehaviour: error on saving cart")
+	}
+
+	return cob.resetPaymentSelectionIfInvalid(ctx, cart)
+}
+
+// UpdateItems updates multiple cart items
+func (cob *InMemoryBehaviour) UpdateItems(ctx context.Context, cart *domaincart.Cart, itemUpdateCommands domaincart.ItemUpdateCommands) (*domaincart.Cart, domaincart.DeferEvents, error) {
+	if !cob.cartStorage.HasCart(cart.ID) {
+		return nil, nil, fmt.Errorf("cart.infrastructure.InMemoryBehaviour: Cannot update - Guestcart with id %v not existent", cart.ID)
+	}
+
+	for _, itemUpdateCommand := range itemUpdateCommands.ItemUpdateCommands {
+		err := cob.updateItem(ctx, cart, itemUpdateCommand)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	err := cob.cartStorage.StoreCart(cart)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "cart.infrastructure.InMemoryBehaviour: error on saving cart")
+	}
+
+	return cob.resetPaymentSelectionIfInvalid(ctx, cart)
+}
+
+func (cob *InMemoryBehaviour) updateItem(ctx context.Context, cart *domaincart.Cart, itemUpdateCommand domaincart.ItemUpdateCommand) error {
```

Comment:
Should we add error handling for delivery for supplied delivery not found? or supplied item id not in delivery? We have vars for those cases in cartServicePort: ErrItemNotFound, ErrDeliveryCodeNotFound etc